### PR TITLE
Updating SML/NJ to the latest version

### PR DESCRIPTION
**Pull request by hoheinzollern**
Hi all, I would like to request this update of SML/NJ as it has some important bug fixes in the compiler.
I tested it and it works fine on my machine (running OS X 10.11).
The only thing I don't know is how to deal with bottles, so I removed them from the package. As I understand there is a bot creating them, but please let me know if I need to do anything on this.

### CODE REVIEW

**Comment by bfontaine** [comment]
> The only thing I don't know is how to deal with bottles, so I removed them from the package.

Please leave them as is; our bot will update them. Thanks!

**Comment by hoheinzollern** [comment]
Ok, I reverted the bottles as they were before. Thank you for the info.

File: Library/Formula/smlnj.rb [hunk 1]
```diff
@@ -12,127 +12,121 @@ class Smlnj < Formula
     sha256 "d65bdd75825461dc6fd3f15f50de7ebe339b0644e7cb21132b0f2cd742cd293f" => :mavericks
     sha256 "ae8daf8491c3b2d6d2b12ce101fba322791591065b69129ff734325a1a1be063" => :mountain_lion
   end
-
+
```
Comment (bfontaine) [inline Library/Formula/smlnj.rb]: There’s a trailing space here.

**Comment by hoheinzollern** [comment]
I sill the build failed on the OSX 10.9.5 with error message:

```
FATAL:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../libexec/as/i386/as: I don't understand 'm' flag!
```

is this a common problem? how can I fix it if I don't have such configuration?

**Comment by dunn** [comment]
Possibly a change in this version broke compatibility with Mavericks; would you mind letting the developers know?

**Comment by hoheinzollern** [comment]
Oh, yes, thanks. It seems like they also noticed the problem:
http://smlnj-gforge.cs.uchicago.edu/tracker/index.php?func=detail&aid=151&group_id=33&atid=215
I'll keep an eye on it and report when that has been solved.
Would it be possible to update only for the newer versions of OSX in the meanwhile? Or do the packages have to be aligned for all versions?

**Comment by dunn** [comment]
We can mark it as failing on Mavericks, but we'll only do that if it looks like the developers aren't going to fix it.  Thanks for taking a look!  We can leave this PR open while we wait for updates.

**Comment by apjanke** [comment]
Duplicate of #49530. Looks like this formula has issues and is heading for the boneyard.

If you can get this fixed, or are sure <= Mavericks is no longer supported, please re-submit on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core).

**Pull request closed without merging**